### PR TITLE
CLN: io/formats/html.py: refactor

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [ n/a] follow-on from #22726
- [x] tests ~~added~~ / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

remove `max_rows` and `max_cols` from `HTMLFormatter` as truncation is now handled by `DataFrameFormatter`

note: this PR should be orthogonal to #22655

### CODE REVIEW

**Review by gfyoung — CHANGES_REQUESTED** [review]
Are these classes in anyway considered user-facing?  If not, LGTM.

cc @jreback @WillAyd

File: pandas/io/formats/html.py [hunk 1]
```diff
@@ -35,12 +35,8 @@ def __init__(self, formatter, classes=None, max_rows=None, max_cols=None,
         self.elements = []
         self.bold_rows = self.fmt.kwds.get('bold_rows', False)
         self.escape = self.fmt.kwds.get('escape', True)
-
-        self.max_rows = max_rows or len(self.fmt.frame)
-        self.max_cols = max_cols or len(self.fmt.columns)
         self.show_dimensions = self.fmt.show_dimensions
-        self.is_truncated = (self.max_rows < len(self.fmt.frame) or
-                             self.max_cols < len(self.fmt.columns))
+        self.is_truncated = self.fmt.is_truncated
```
Comment (jreback) [inline pandas/io/formats/html.py]: this could be a property?

File: pandas/io/formats/html.py [hunk 2]
```diff
@@ -318,9 +311,8 @@ def _write_body(self, indent):
         self.write('<tbody>', indent)
         indent += self.indent_delta
 
-        fmt_values = {}
-        for i in range(min(len(self.columns), self.max_cols)):
-            fmt_values[i] = self.fmt._format_col(i)
+        ncols = len(self.fmt.tr_frame.columns)
```
Comment (jreback) [inline pandas/io/formats/html.py]: these could also be properties fro clarify

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
these changes are ok, though @simonjayhawkins pls add a note anyhow in api breaking changes (they are private but just a notice)

**Comment by simonjayhawkins** [comment]
@gfyoung @jreback note about api breaking change has been added to whatsnew

**Review by gfyoung — APPROVED** [review]
cc @jreback

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Pull request merged**